rust/azure_iot_operations_mqtt/src/session/receiver.rs

```diff
@@ -1134,7 +1134,7 @@ mod tests {
         ); // Type 4
 
         // Register a new filter again
-        let topic_filter8 = topic_filter7.clone(); // Type 4
```

Comment:
I would prefer to exclude this change for clarity reasons - I understand it is technically unnecessary, but having this clone reduces cognitive overhead in test maintenance.

Reply:
there were a few others in this file that clippy detected that needed to stay as well - I'll revert this one then!